src/dev/build/tasks/nodejs_modules/clean_client_modules_on_dll_task.js

```diff
@@ -66,10 +66,38 @@ export const CleanClientModulesOnDLLTask = {
     // side code entries that were provided
     const serverDependencies = await getDependencies(baseDir, serverEntries);
 
+    // This fulfill a particular exceptional case where
+    // we need to keep loading a file from a node_module
+    // only used in the front-end like we do when using the file-loader
+    // in https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/maps/public/connected_components/map/mb/view.js
+    //
+    // manual list of exception modules
+    const manualExceptionModules = [
+      'mapbox-gl'
```

Comment:
Are we not able to exclude just the specific module that we depend on from `mapbox-gl`?

Reply:
In that case we are because looks like the `mapbox-gl` dist files doesn't declare other requires, I just tried to build a system for other future needs on that matter of manual exclusions. Do you prefer to simplify it for now?

Reply:
Eh, it's up to you but when I red this I figured it meant that it would keep all the files in the `mapbox-gl` module rather than just the one that was imported.

Reply:
Ah I see your question now. However the current logic only allow to clear an entire module according those rules https://github.com/elastic/kibana/blob/6f1253620b817c2fd54ef02978b0c836b135562f/src/dev/build/tasks/nodejs_modules/webpack_dll.js#L69 or doesn't clear it at all. So maybe instead of just changing that entire logic right now, we would be better to just allow that module to not be cleared